### PR TITLE
Chore/startup

### PR DESCRIPTION
**Pull request by frickjack**
* lazy initialization of access tokens and clients
* update to codecept 2, webdriver 5

### CODE REVIEW

File: suites/apis/dbgapTest.js [hunk 1]
```diff
@@ -155,10 +154,25 @@ Scenario('dbGaP Sync: created signed urls (from s3 and gs) to download, try crea
...
+    let phs000178s3FileContents = null;
+    let phs000178gsFileContents = null;
+
+    try {
+      phs000178s3FileContents = await fence.do.getFileFromSignedUrlRes(
+        signedUrls3phs000178Res);
+    } catch (err) {
+      let url = signedUrls3phs000178Res && signedUrls3phs000178Res.data && signedUrls3phs000178Res.data.url;
+      console.log(`Failed to fetch signed url ${url}`, err);
+      chai.expect(false, `Failed to fetch signed url ${url}`).to.be.true;
```
Comment (paulineribeyre) [inline suites/apis/dbgapTest.js]:
```suggestion
      throw Error(`Failed to fetch signed url ${url}`);
```

File: suites/apis/dbgapTest.js [hunk 2]
```diff
@@ -155,10 +154,25 @@ Scenario('dbGaP Sync: created signed urls (from s3 and gs) to download, try crea
...
+    }
+    try {
+      phs000178gsFileContents = await fence.do.getFileFromSignedUrlRes(
+        signedUrlgsPhs000178Res);
+    } catch (err) {
+      let url = signedUrlgsPhs000178Res && signedUrlgsPhs000178Res.data && signedUrlgsPhs000178Res.data.url;
+      console.log(`Failed to fetch signed url ${url}`, err);
+      chai.expect(false, `Failed to fetch signed url ${url}`).to.be.true;
```
Comment (paulineribeyre) [inline suites/apis/dbgapTest.js]:
```suggestion
      throw Error(`Failed to fetch signed url ${url}`);
```

File: suites/apis/getCoremetadataTest.js [hunk 3]
```diff
@@ -35,9 +38,11 @@ BeforeSuite(async (nodes, sheepdog) => {
 
   valid_file = nodes.getFileNode().clone();
   await sheepdog.complete.addNode(valid_file);
-
+  //console.log('Got file', valid_file);
+  expect(!! valid_file.did).to.be.true;
```
Comment (paulineribeyre) [inline suites/apis/getCoremetadataTest.js]:
```suggestion
  expect(!! valid_file.did, "Sheepdog added data_file without adding did").to.be.true;
```

File: services/apis/fence/fenceProps.js [hunk 4]
```diff
@@ -1,19 +1,77 @@
...
 
+/**
+ * Runs a fence command for creating a client
+ * @param {string} clientName - client name
+ * @param {string} userName - user name
+ * @param {string} clientType - client type (implicit or basic)
+ * @param {string} arboristPolicies - space-delimited list of arborist policies to give to client
+ * @returns {json}
+ */
+function createClient(clientName, userName, clientType, arboristPolicies=null) {
```
[the remark this answers is not in the document]
Comment (frickjack) [reply services/apis/fence/fenceProps.js]: this function is private to the module.

File: utils/apiUtil.js [hunk 5]
```diff
@@ -45,29 +45,37 @@ function parseFenceError(responseBody) {
...
-      this.requestHeaders = request.headers;
-      this.requestBody = request.body;
-      this.requestURL = request.uri.href;
-    } catch (e) {
-      // ignore if missing request attribute
-      console.log('Gen3Response could not parse expected `request` attributes like `method`, `headers`, `body`, etc.')
-      console.log(`Gen3Response got request: ${request}`)
+    this.parsedFenceError = (isErrorPage(data) ? parseFenceError(data) : undefined);
+    this.body = this.parsedFenceError ? undefined : (data || body); // include body if not error page
+    this.data = data || body;
```
[the remark this answers is not in the document]
Comment (frickjack) [reply utils/apiUtil.js]: `.body` is still used multiple places - including in the expected results comparison

File: utils/apiUtil.js [hunk 6]
```diff
@@ -45,29 +45,37 @@ function parseFenceError(responseBody) {
...
-   * @param {number} statusCode
+   * @param {number|Object} status or {status}
    * @param {Object} [request]
    * @param {string} [request.method]
    * @param {string} [request.headers]
    * @param {string} [request.uri.href]
    * @param {string} [fenceError] Parsed from body if determined to be an error page
    */
-  constructor({ body, statusCode, request, fenceError }) {
+  constructor({ data, body, status, request, fenceError }) {
```
[the remark this answers is not in the document]
Comment (frickjack) [reply utils/apiUtil.js]: ok

File: services/apis/indexd/indexdTasks.js [hunk 7]
```diff
@@ -79,12 +79,16 @@ module.exports = {
     //console.log("indexd addFileIndices waiting on promiseList of length: " + promiseList.length, promiseList);
     // This Promise.all trick does not work for some reason - ugh!
     // Have to figure it out later
-    const success = await Promise.all(promiseList).then(
-      () => true,
-      (v) => {
-        return false;
+    const success = await (
+      async () => {
```
[the remark this answers is not in the document]
Comment (frickjack) [reply services/apis/indexd/indexdTasks.js]: it doesn't fix it - don't know what's up - added a clarifying note

**Pull request merged**